Sources/MapboxCoreNavigation/RouteController.swift

```diff
@@ -254,23 +254,33 @@ open class RouteController: NSObject {
                   completion?(.failure(RouteControllerError.failedToSerializeRoute))
                   return
         }
-
-        let routeRequest = Directions().url(forCalculating: progress.routeOptions).absoluteString
-        let routes = Routes(routesResponse: routeJSONString,
-                            routeIndex: 0,
-                            legIndex: UInt32(progress.legIndex),
-                            routesRequest: routeRequest)
-
-        sharedNavigator.setRoutes(routes, uuid: sessionUUID) { result in
-            completion?(result)
+        
+        DispatchQueue.global().async {
+            let routeRequest = Directions().url(forCalculating: progress.routeOptions).absoluteString
+            
+            let parsedRoutes = RouteParser.parseDirectionsResponse(forResponse: routeJSONString,
```

Comment:
Explicitly moved to a background thread as this operation may take significant time. NN is planning to provide an async analog for this function, so we'll be able to update this part later.

Reply:
In the end I've decided to not bother switching to a background thread. This move creates race condition between setting the route and processing location updates which results in Navigator skipping status updates and tests failing. I've also measured parse time and found that long route (from US East cost to West coast) takes approximately 0.11 seconds to complete, which I think is good enough for us to wait until NN introduces their own async version of the method. I believe NN's async `parseDirectionsResponse` may work better because it shall use the same `worker thread` as used by Navigator, so that would eliminate the race condition introduced here.